src/many-modules/src/_9_account/features/multisig.rs

```diff
@@ -12,9 +12,48 @@ use many_types::ledger::TokenAmount;
 use many_types::Timestamp;
 use minicbor::bytes::ByteVec;
 use minicbor::data::Type;
-use minicbor::{encode, Decode, Decoder, Encode, Encoder};
+use minicbor::{encode, decode, Decode, Decoder, Encode, Encoder};
 use std::collections::{BTreeMap, BTreeSet};
 
+/// A short note in a transaction
+#[derive(Clone, Debug, PartialEq)]
+// Using AsRef<str> for extensibility:
+// We might want to borrow a &Memo<str> instead of a &Memo<String>
+pub struct Memo<S: AsRef<str>>(S);
```

Comment:
Cow would be even better, no? That way it can be owned.

Reply:
You're right, that's a great idea.

Reply:
@hansl the `Cow` idea isn't really working out because even an owned `Cow` needs a lifetime parameter, and I cannot have an impl for `'static` and an arbitrary `'a`, so the `Cow` lifetime will propagate up and the `SubmitTransactionsArgs` will need a lifetime parameter, which means we'd also need to modify how the `define_events!` macro works. My solution is cleaner than the `Cow`. Nevertheless, with my solution you *can* have a `Memo<Cow<'a, str>>`, because `Cow` implements `AsRef`.